alacritty_terminal/src/input.rs

```diff
@@ -445,50 +484,45 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
-    /// Underline URLs and change the mouse cursor when URL hover state changes.
-    fn update_url_highlight(&mut self, point: Point, modifiers: ModifiersState) {
-        let mouse_mode =
-            TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG | TermMode::MOUSE_REPORT_CLICK;
+    fn url_bounds_at_point(&self, url: Url, point: Point) -> RangeInclusive<Linear> {
+        let Url { origin, text } = url;
+        let cols = self.ctx.size_info().cols().0;
 
-        // Only show URLs as launchable when all required modifiers are pressed
-        let url = if self.mouse_config.url.mods().relaxed_eq(modifiers)
-            && (!self.ctx.terminal().mode().intersects(mouse_mode) || modifiers.shift)
-            && self.mouse_config.url.launcher.is_some()
-        {
-            self.ctx.terminal().url_search(point.into())
+        // Calculate the URL's start position
+        let lines_before = (origin + cols - point.col.0 - 1) / cols;
+        let (start_col, start_line) = if lines_before > point.line.0 {
+            (0, 0)
         } else {
-            None
+            let start_col = (cols + point.col.0 - origin % cols) % cols;
+            let start_line = point.line.0 - lines_before;
+            (start_col, start_line)
         };
 
-        if let Some(Url { origin, text }) = url {
-            let cols = self.ctx.size_info().cols().0;
-
-            // Calculate the URL's start position
-            let lines_before = (origin + cols - point.col.0 - 1) / cols;
-            let (start_col, start_line) = if lines_before > point.line.0 {
-                (0, 0)
-            } else {
-                let start_col = (cols + point.col.0 - origin % cols) % cols;
-                let start_line = point.line.0 - lines_before;
-                (start_col, start_line)
-            };
-            let start = Point::new(start_line, Column(start_col));
+        let start = Point::new(start_line, Column(start_col));
 
-            // Calculate the URL's end position
-            let len = text.width();
-            let end_col = (point.col.0 + len - origin) % cols - 1;
```

Comment:
There was a bug, when url ends at the last column there was an unsigned overflow (0 - 1).